tests/integration/test_col_functions.py

```diff
@@ -102,21 +148,21 @@ def test_col_is_not_in_list(spark):
 
 
 def test_col_sql_expression(spark):
-    test_df = spark.createDataFrame([["str1", 1, 1], ["str2", None, None], ["", 2, 3]], SCHEMA + ", c: string")
+    test_df = spark.createDataFrame([["str1", 1, 1], ["str2", None, None], ["", 3, 2]], SCHEMA + ", c: string")
 
     actual = test_df.select(
-        sql_expression("a = 'str2'"),
-        sql_expression("b is null", name="test", negate=True),
-        sql_expression("c is null", msg="failed validation", negate=True),
-        sql_expression("b < c", msg="b is greater or equal c", negate=False),
+        sql_expression("a = 'str1'"),
+        sql_expression("b is not null", name="test", negate=True),
+        sql_expression("c is not null", msg="failed validation", negate=True),
+        sql_expression("b > c", msg="b is not greater than c", negate=True),
```

Comment:
please revert the change for the `sql_expression`. We have updated this logic recently. The test should not be changed.